File: Q44/main.c
/*
An element is a leader if it is greater than or equal to all elements to its right.
The rightmost element is always a leader.
*/

#include<stdio.h>

int main(){

    int n;
    scanf("%d",&n);

    int arr[n],leaders[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }

    int mfr = arr[n-1]; //maximum element from right
    int ind = 0;
             
    leaders[ind++] = mfr;

    for(int i=n-2;i>=0;i--){
        if(arr[i]>=mfr){
            mfr = arr[i];
            leaders[ind++] = mfr;
        }
    }

    for(int i=ind-1;i>=0;i--){
        printf("%d ",leaders[i]);
    }

    return 0;
}

File: Q44/main.py
n = int(input())

arr=[]

for i in range(n):
    arr.append(int(input()))

leaders=[]

mfr=arr[n-1]
leaders.append(mfr)

for i in range(n-2,0,-1):
    if arr[i]>=mfr:
        mfr = arr[i]
        leaders.append(mfr)

for i in range(len(leaders)-1,-1,-1):
    print(leaders[i],end=" ")